### PR TITLE
feat(materials): Add support for ZAP reports

### PR DESCRIPTION
**Pull request by javirln**
This patch introduces support for a new material type called ZAP_DAST_ZIP, which represents a ZAP DAST report in ZIP format. The crafter will validate the incoming file as a valid ZIP archive and, using the extracted JSON report, ensure that the program used is ZAP.

Refs: #1359

### CODE REVIEW

File: app/controlplane/api/workflowcontract/v1/crafting_schema_validations.go [hunk 1]
```diff
@@ -30,6 +30,7 @@ import (
 // detection. The order of the list is important as it defines the order of the
 // detection process. Normally from most common one to the least common one and weaker validation method.
 var CraftingMaterialInValidationOrder = []CraftingSchema_Material_MaterialType{
+	CraftingSchema_Material_ZAP_DAST_ZIP,
```
Comment (migmartri) [inline app/controlplane/api/workflowcontract/v1/crafting_schema_validations.go:33]: why first? Isn't it too expensive? I'd make sure we order by less expensive to more

File: pkg/attestation/crafter/materials/zap.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+	return &ZAPCrafter{
+		backend:       backend,
+		crafterCommon: &crafterCommon{logger: l, input: materialSchema},
+	}, nil
+}
+
+// Craft will extract the ZAP JSON report from the zip file and upload it to the CAS
+func (i *ZAPCrafter) Craft(ctx context.Context, filePath string) (*api.Attestation_Material, error) {
+	archive, err := zip.OpenReader(filePath)
```
Comment (migmartri) [inline pkg/attestation/crafter/materials/zap.go:65]: could we check if it's a zip file?

File: app/controlplane/api/workflowcontract/v1/crafting_schema_validations.go (hunk 1 above)
Comment (javirln) [reply app/controlplane/api/workflowcontract/v1/crafting_schema_validations.go:33]: Yes, I don't know why I placed it there, it should be lower on the list.

File: pkg/attestation/crafter/materials/zap.go (hunk 2 above)
Comment (javirln) [reply pkg/attestation/crafter/materials/zap.go:65]: That OpenReader checks already that the incoming file is a zip file but we can do a manual check in any case.

**Review by migmartri — COMMENTED** [review]
ptal at my comment

**Pull request merged**